rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/tsdb/20_mapping.yml

```diff
@@ -312,7 +312,7 @@ nested dimensions:
       reason: message changed in 8.2.0
 
   - do:
-      catch: /cannot have nested fields when index is in \[index.mode=time_series\]/
+      catch: /time_series_dimension can't be configured in nested field \[nested.dim\]/
```

Comment:
this PR will change the nested failed message. the field dimension check is before tsdb nested check.